src/rest_api/video_application.py

```diff
@@ -0,0 +1,161 @@
+        required=True),
+    'TimeRange': fields.Nested(
+        timerange_model,
+        description='The model for the time range analyzed by the program'),
+    'SourcePath': fields.String(
+        description='The path of the video to analyze',
+        required=True),
+    'User': fields.String(
+        description='The requesting user',
+        required=True),
+})
+
+
+@api.route('/video')
+class AnalyzeVideo(Resource):
+    @api.expect(video_fields)
+    def post(self) -> object:
+        """Analyzes a video and outputs the data to the database
+
+        Returns
+        -------
+        object
+            A dummy json response to confirm that the analysis has begun
+        """
+
+        data = request.get_json()
+
+        video_id = data['ID']
+        video_path = data['SourcePath']
+        video_time_range = data['TimeRange']
+
+        if video_time_range["From"] == video_time_range["To"]:
+            api.abort(
+                400,
+                "From and to can not be equal"
+            )
+
+        if not os.path.isfile(video_path):
+            api.abort(
+                400,
+                "The given source path '{}' does not seem to exist"
+                .format(video_path)
+            )
+
+        _thread.start_new_thread(
+            process_data_and_extract_emotions,
+            (
+                video_id,
+                video_path,
+                video_time_range
+            )
+        )
+
+        return {'Response': 'The request has been sent and should be'
+                            ' updated in Splunk as soon as it is done.'}, 201
+
+
+@api.route('/video_with_audio')
+class AnalyzeVideoWithSong(Resource):
+    @api.expect(video_fields_with_song)
+    def post(self) -> object:
+        """Analyzes a video together with a song and outputs the data to the database
+
+        Returns
+        -------
+        object
+            A dummy json response to confirm that the analysis has begun
+        """
+
+        data = request.get_json()
+
+        video_id = data['ID']
+        song_id = data['SongID']
+        video_path = data['SourcePath']
+        video_time_range = data['TimeRange']
+
+        if video_time_range["From"] == video_time_range["To"]:
+            api.abort(
+                400,
+                "From and to can not be equal"
+            )
+
+        if not os.path.isfile(video_path):
+            api.abort(
+                400,
+                "The given source path '{}' does not seem to exist"
+                .format(video_path)
+            )
+
+        _thread.start_new_thread(
+            process_data_and_extract_emotions_with_song,
+            (
+                video_id,
+                video_path,
+                video_time_range,
+                song_id
+            )
+        )
+
+        return {'Response': 'The request has been sent and should be'
+                            ' updated in Splunk as soon as it is done.'}
```

Comment:
I think we should update this to not include splunk.

Reply:
This can be removed when the actual endpoints are being finished :)